azure/scope/machinepool.go

```diff
@@ -510,12 +507,12 @@ func (m *MachinePoolScope) PatchObject(ctx context.Context) error {
 
 // Close the MachineScope by updating the machine spec, machine status.
 func (m *MachinePoolScope) Close(ctx context.Context) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "scope.MachinePoolScope.Close")
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "scope.MachinePoolScope.Close")
```

Comment:
Yes'ish, because the `ctx` that is returned is a new `ctx` struct which may now contain a correlationID, span, and other values in the bag. We could change the name of the new context, but this ensure that the original is not used below. Only the new `ctx` would be used. Use of the original is in nearly all cases an error.

Reply:
> Use of the original is in nearly all cases an error.

Got it. Thanks for the clarification.